supertokens_python/recipe/session/session_class.py

```diff
@@ -66,20 +78,40 @@ async def update_access_token_payload(
         if user_context is None:
             user_context = {}
 
-        response = await self.recipe_implementation.regenerate_access_token(
+        result = await self.recipe_implementation.regenerate_access_token(
             self.access_token, new_access_token_payload, user_context
         )
-        if response is None:
+        if result is None:
             raise_unauthorised_exception("Session does not exist anymore.")
 
-        self.access_token_payload = response.session.user_data_in_jwt
-        if response.access_token is not None:
-            self.access_token = response.access_token.token
-            self.new_access_token_info = {
-                "token": response.access_token.token,
-                "expiry": response.access_token.expiry,
-                "createdTime": response.access_token.created_time,
-            }
+        self.access_token_payload = result.session.user_data_in_jwt
+        if result.access_token is not None:
+            self.access_token = result.access_token.token
+
+            # new_access_token_info = {
+            #     "token": result.access_token.token,
+            #     "expiry": result.access_token.expiry,
+            #     "createdTime": result.access_token.created_time,
+            # }
+            self.response_mutators.append(
+                partial(
+                    set_front_token_in_headers,
+                    user_id=self.user_id,
+                    expires=result.access_token.expiry,
+                    jwt_payload=self.access_token_payload,  # TODO: Check if this is correct
+                )
+            )
+            self.response_mutators.append(
+                partial(
+                    set_token,
+                    response=None,
```

Comment:
We are allowing it to pass `None` so that we can add function calls to `response_mutators` and later on pass the actual response object later on in the `mange_session_post_response` function.

Reply:
I get that, but the partial call right above this doesn't set the response to None. If that works, I don't see why it's required here.